### PR TITLE
expose sslContext and prefer tlsPort if present

### PR DESCRIPTION
**Pull request by pjain1**
Fixes https://github.com/druid-io/tranquility/issues/255
Depends on https://github.com/metamx/scala-util/pull/37

### CODE REVIEW

**Comment by himanshug** [comment]
:+1: let us update scala-util dependency version too as soon as https://github.com/metamx/scala-util/pull/37 is done and released.

**Comment by pjain1** [comment]
Travis will fail till scala-util dependency is updated after metamx/scala-util#37 is released

**Comment by pjain1** [comment]
@gianm changed to use `Option`, please verify if it is indeed what you expected. I have tested it locally with https enabled on both Overlord and MiddleManager.

File: core/src/main/scala/com/metamx/tranquility/finagle/FinagleRegistry.scala [hunk 1]
```diff
@@ -87,7 +87,8 @@ class FinagleRegistry(
       .logger(FinagleLogger)
       .daemon(true)
       .failFast(config.finagleEnableFailFast)
-      .build()
+    val client = if(config.sslContextOption.isDefined) clientBuilder.tls(config.sslContextOption.get).build() else clientBuilder.build()
```
Comment (gianm) [inline core/src/main/scala/com/metamx/tranquility/finagle/FinagleRegistry.scala]: More scala-y:

```scala
val client = config.sslContextOption match {
  case Some(sslContext) => clientBuilder.tls(sslContext).build()
  case None => clientBuilder.build()
}
```

Or even:

```scala
config.sslContextOption.foreach(sslContext => clientBuilder.tls(sslContext))
val client = clientBuilder.build()
```

**Comment by pjain1** [comment]
Trying to fix issues with with overlord based task discovery

**Comment by pjain1** [comment]
@gianm can you have a look again

**Comment by himanshug** [comment]
:+1:

**Pull request merged**